sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Filtering/Filter.cs

```diff
@@ -0,0 +1,719 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Monitor.OpenTelemetry.LiveMetrics.Internals.Filtering
```

Comment:
Do you want to use #nullable disable to minimize the change from AI SDK?

Reply:
It's not too many changes, except for one or two tricky cases.

Reply:
I'll disable the nullable and minimal the change to avoid the tricky case for now.